modules/ossm-extensions-migrating-to-wasmplugin.adoc

```diff
@@ -0,0 +1,54 @@
+////
+This module included in the following assemblies:
+*service_mesh_/v2x/ossm-extensions.adoc
+////
+:_content-type: PROCEDURE
+[id="ossm-extensions-migrating-to-wasmplugin_{context}"]
+= Migrating to `WasmPlugin` resources
+
+To upgrade your WebAssembly extensions from the `ServiceMeshExtension` API to the `WasmPlugin` API, you remove the old plugin and create a new plugin file.  Creating an extension file involves revising the extension and compiling it into a WebAssebmly module, and then re-deploying it into a container.
+
+
+.Prerequisites
+
+* `ServiceMeshControlPlane` upgraded to version 2.2 or later.
+
+[CAUTION]
+====
+Because both plugins will be invoked for every request, you might want to remove your existing  `ServiceMeshExtension` resource before creating the new `WasmPlugin` resource.  You might get undesired results having two plugins active at the same time.
+====
+
+.Procedure
+
+. Create a backup of your `ServiceMeshExtension` resource.
+
+. Update the extension for the API changes.
+
+.. Move your resource to the root of the container file system if it is not already there.  Extension files must live in the root of the container filesystem.
+
+.. Rename your resource to `plugin.wasm` if that is not already the name. Extension files must be named `plugin.wasm`.
+
+.. Update the fields in your `plugin.wasm` file to point to the fields in the new API.
```

Comment:
Rename this to: Optionally, remove the `manifest.yaml` file from your container. It's not necessary anymore, but it will not cause any harm if present. It's just ignored.